### PR TITLE
Feat playwright api

### PR DESCRIPTION
**Pull request by OpalinskyiSerhii98**

### CODE REVIEW

File: src/components/Body.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import BaseComponent from "./BaseComponent.js";
+
+
+export default class Body extends BaseComponent {
+    constructor(page) {
+        super(page, page.locator('body'));
+        this.signUpButton =  page.locator('button', {hasText: 'Sign Up'})
+    }
+}
```
Comment (sttaran) [inline src/components/Body.js]: You should associate your components with components on a page but not with HTML

Comment (sttaran) [reply src/components/Body.js]: I do not think you need it

File: tests/garage/garage.spec.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+test.describe('Garage', ()=>{
+    test('create car', async ({garagePage, page})=>{
+        await expect(garagePage.addCarButton).toBeVisible()
+
+        await garagePage.addCarButton.click()
+
+        await page.locator("#addCarBrand").selectOption('Audi')
+        await page.locator("#addCarModel").selectOption('TT')
+        await page.locator("#addCarMileage").fill('180')
+        await page.locator('.modal-content  .btn-primary').click()
```
Comment (sttaran) [inline tests/garage/garage.spec.js]: use POM

**Review by sttaran — CHANGES_REQUESTED** [review]

File: tests/garage/profile.spec.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+test.describe('Profile', ()=>{
+    test('open profile tab', async ({page})=>{
+
+        await page.route('https://qauto.forstudy.space/api/users/profile', (route)=>{
+            return route.fulfill({
+                status: 200,
+                body: JSON.stringify(PROFILE_MOCK_RESPONSE)
+            })
+        })
+        await page.goto('/panel/profile')
```
Comment (sttaran) [inline tests/garage/profile.spec.js:14]: you should check that mocked value is displayed on UI

**Review by sttaran — CHANGES_REQUESTED** [review]

File: tests/garage/profile.spec.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+        const profileName = await page.$eval('.profile_name', (element) => element.textContent.trim());
```
Comment (sttaran) [inline tests/garage/profile.spec.js]: use https://playwright.dev/docs/api/class-locatorassertions#locator-assertions-to-have-text

and use POM

**Review by sttaran — CHANGES_REQUESTED** [review]

File: tests/garage/profile.spec.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+        await page.route('https://qauto.forstudy.space/api/users/profile', (route)=>{
+            return route.fulfill({
+                status: 200,
+                body: JSON.stringify(PROFILE_MOCK_RESPONSE)
+            })
+        })
+        await page.goto('/panel/profile')
+
+        const profileName = page.locator('.profile_name')
+        const name = "John Doe"
```
Comment (sttaran) [inline tests/garage/profile.spec.js]: create name based on PROFILE_MOCK_RESPONSE

**Review by sttaran — CHANGES_REQUESTED** [review]

File: tests/garage/profile.spec.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+        await page.route('https://qauto.forstudy.space/api/users/profile', (route)=>{
+            return route.fulfill({
+                status: 200,
+                body: JSON.stringify(PROFILE_MOCK_RESPONSE)
+            })
+        })
+        await page.goto('/panel/profile')
+
+        const profileName = page.locator('.profile_name')
```
Comment (sttaran) [inline tests/garage/profile.spec.js]: use POM

**Review by sttaran — CHANGES_REQUESTED** [review]

File: tests/garage/profile.spec.js (hunk 5 above)
Comment (OpalinskyiSerhii98) [reply tests/garage/profile.spec.js]: done

**Pull request merged**